File: business/src/main/java/com/isystk/sample/common/helper/UserHelper.java
package com.isystk.sample.common.helper;

import com.isystk.sample.common.exception.NoDataFoundException;
import com.isystk.sample.common.util.DateUtils;
import com.isystk.sample.domain.dao.AuditInfoHolder;
import com.isystk.sample.domain.dao.TUserDao;
import com.isystk.sample.domain.dto.TUserCriteria;
import com.isystk.sample.domain.entity.TUser;
import java.util.List;
import org.slf4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

/**
 * ユーザーヘルパー
 */
@Component
public class UserHelper {

  private static final Logger log = org.slf4j.LoggerFactory.getLogger(UserHelper.class);
  @Autowired
  TUserDao tUserDao;

  /**
   * ユーザーを全件取得します。
   *
   * @return
   */
  public List<TUser> getUserList() {
    TUserCriteria criteria = new TUserCriteria();
    criteria.setDeleteFlgFalse(true);
    return tUserDao.findAll(criteria);
  }

  /**
   * ユーザーを取得します。
   *
   * @return
   */
  public TUser getUser(Integer userId) {
    TUserCriteria criteria = new TUserCriteria();
    criteria.setUserIdEq(userId);
    return tUserDao.findOne(criteria).orElseThrow(
        () -> new NoDataFoundException("userId=" + userId + "のデータが見つかりません。"));
  }

  /**
   * ログインユーザーを取得します。
   *
   * @return
   */
  public Integer getLoginUserId() {
    return getUser().getUserId();
  }

  /**
   * ログインユーザーを取得します。
   *
   * @return
   */
  public TUser getUser() {
    TUserCriteria criteria = new TUserCriteria();
    criteria.setEmailEq(AuditInfoHolder.getAuditUser());
    return tUserDao.findOne(criteria).orElseThrow(
        () -> new NoDataFoundException(
            "email=" + AuditInfoHolder.getAuditUser() + "のデータが見つかりません。"));
  }

  /**
   * 最終ログイン日時を更新します。
   *
   * @return
   */
  public void updateLastLogin() {
    TUser tUser = getUser();
    tUser.setLastLoginTime(DateUtils.getNow());
    tUserDao.update(tUser);
  }

}

File: business/src/main/kotlin/com/isystk/sample/common/helper/UserHelper.kt
package com.isystk.sample.common.helper

import com.isystk.sample.common.exception.NoDataFoundException
import com.isystk.sample.common.util.DateUtils
import com.isystk.sample.domain.dao.AuditInfoHolder
import com.isystk.sample.domain.dao.UserDao
import com.isystk.sample.domain.dto.UserCriteria
import com.isystk.sample.domain.entity.User
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import java.math.BigInteger

/**
 * ユーザーヘルパー
 */
@Component
class UserHelper {
    @Autowired
    var userDao: UserDao? = null

    /**
     * ユーザーを全件取得します。
     *
     * @return
     */
    val userList: List<User>
        get() {
            val criteria = UserCriteria()
            criteria.deleteFlgFalse = true
            return userDao!!.findAll(criteria)
        }

    /**
     * ユーザーを取得します。
     *
     * @return
     */
    fun getUser(userId: Long): User {
        val criteria = UserCriteria()
        criteria.idEq = userId
        return userDao!!.findOne(criteria) ?: throw NoDataFoundException("userId=" + userId + "のデータが見つかりません。")
    }

    /**
     * ログインユーザーを取得します。
     *
     * @return
     */
    val loginUserId: Long
        get() = user.id!!

    /**
     * ログインユーザーを取得します。
     *
     * @return
     */
    val user: User
        get() {
            val criteria = UserCriteria()
            criteria.emailEq = AuditInfoHolder.auditUser
            return userDao!!.findOne(criteria) ?: throw
                NoDataFoundException(
                        "email=" + AuditInfoHolder.auditUser + "のデータが見つかりません。")
        }

    /**
     * 最終ログイン日時を更新します。
     *
     * @return
     */
    fun updateLastLogin() {
        val user = user
        user.lastLoginAt = DateUtils.now
        userDao!!.update(user)
    }

    companion object {
        private val log = LoggerFactory.getLogger(UserHelper::class.java)
    }
}